File: SDKs/amazonaws.com/cognito-idp/2016-04-18/python/src/sdk/models/shared/compromisedcredentialsriskconfigurationtype.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import compromisedcredentialsactionstype as shared_compromisedcredentialsactionstype
from ..shared import eventfiltertype_enum as shared_eventfiltertype_enum


@dataclass_json
@dataclasses.dataclass
class CompromisedCredentialsRiskConfigurationType:
    r"""CompromisedCredentialsRiskConfigurationType
    The compromised credentials risk configuration type.
    """
    
    actions: shared_compromisedcredentialsactionstype.CompromisedCredentialsActionsType = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Actions') }})
    event_filter: Optional[list[shared_eventfiltertype_enum.EventFilterTypeEnum]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EventFilter') }})
    

File: SDKs/amazonaws.com/cognito-idp/2016-04-18/java/lib/src/main/java/openapisdk/models/shared/CompromisedCredentialsRiskConfigurationType.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CompromisedCredentialsRiskConfigurationType
 * The compromised credentials risk configuration type.
**/
public class CompromisedCredentialsRiskConfigurationType {
    @JsonProperty("Actions")
    public CompromisedCredentialsActionsType actions;
    public CompromisedCredentialsRiskConfigurationType withActions(CompromisedCredentialsActionsType actions) {
        this.actions = actions;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("EventFilter")
    public EventFilterTypeEnum[] eventFilter;
    public CompromisedCredentialsRiskConfigurationType withEventFilter(EventFilterTypeEnum[] eventFilter) {
        this.eventFilter = eventFilter;
        return this;
    }
}